src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.csproj

```diff
@@ -20,84 +18,27 @@
+    <TaskProject>Microsoft.AspNetCore.Razor.Tasks</TaskProject>
   </PropertyGroup>
 
-  <Target Name="_BuildDependencyProjects">
-    <!-- 
-      The Microsoft.AspNetCore.Razor.Tasks package needs to support both net46 and netstandard2.0 for desktop vs
-      coreclr MSBuild - so we have to build it twice.
-
-      We're careful here to avoid setting properties when building the other projects. This can create problems
-      with concurrency.
-
-      First, build the project, then copy it to the ouput directory, then add it as packable content.
-    -->
-    <MSBuild Projects="@(TaskProject)" />
-
-    <MSBuild Projects="@(TaskProject)" Properties="TargetFramework=net46" Targets="BuiltProjectOutputGroup">
-      <Output TaskParameter="TargetOutputs" ItemName="TaskAssemblyNet46" />
-    </MSBuild>
-
-    <MSBuild Projects="@(TaskProject)" Properties="TargetFramework=net46" Targets="DebugSymbolsProjectOutputGroup">
-      <Output TaskParameter="TargetOutputs" ItemName="TaskSymbolNet46" />
-    </MSBuild>
-
-    <MSBuild Projects="@(TaskProject)" Properties="TargetFramework=netstandard2.0" Targets="BuiltProjectOutputGroup">
-      <Output TaskParameter="TargetOutputs" ItemName="TaskAssemblyNetStandard" />
-    </MSBuild>
-
-    <MSBuild Projects="@(TaskProject)" Properties="TargetFramework=netstandard2.0" Targets="DebugSymbolsProjectOutputGroup">
-      <Output TaskParameter="TargetOutputs" ItemName="TaskSymbolNetStandard" />
-    </MSBuild>
-
-    <Copy SourceFiles="@(TaskAssemblyNet46)" DestinationFolder="$(OutputPath)\tasks\net46\">
-      <Output TaskParameter="CopiedFiles" ItemName="FileWrites" />
-    </Copy>
-    <Copy SourceFiles="@(TaskAssemblyNetStandard)" DestinationFolder="$(OutputPath)\tasks\netstandard2.0\">
-      <Output TaskParameter="CopiedFiles" ItemName="FileWrites" />
-    </Copy>
-
-    <Error Text="TaskAssemblyNet46 is empty. This is a bug" Condition="'@(TaskAssemblyNet46)'==''" />
-    <Error Text="TaskAssemblyNetStandard is empty. This is a bug" Condition="'@(TaskAssemblyNetStandard)'==''" />
-
-    <!--
-      Next we need to build the netcoreapp2.0 tools. In this case we need to do a publish, because we need
-      all of the output to put in the package. 
-    -->
-    <RemoveDir Directories="tools\" />
-    <MSBuild Projects="@(ToolProject)" />
-    <MSBuild Projects="@(ToolProject)" Properties="PublishDir=$(MSBuildProjectDirectory)\$(OutputPath)tools\" Targets="Publish" />
-
-    <ItemGroup>
-      <_RazorTool Include="$(OutputPath)tools\**\*" />
-    </ItemGroup>
-
-    <Error Text="_RazorTool is empty. This is a bug" Condition="'@(_RazorTool)'==''" />
-  </Target>
+  <ItemGroup>
+    <ProjectReference Include="..\$(TaskProject)\$(TaskProject).csproj" ReferenceOutputAssembly="false" />
+    <ProjectReference Include="$(ToolProject)" ReferenceOutputAssembly="false" Condition="'$(TargetFramework)'=='netcoreapp2.0'" />
```

Comment:
What happens when tfm is net46?

Reply:
Nothing really. We just need a way to tell MSBuild that Razor.Tools needs to be built before Razor.Design. We don't use it's output and it doesn't get used in the inner build. All of the interesting work happens in the OuterBuild. I could change this to `'$(TargetFramework)'!='net461'` to reflect this better.

Reply:
I tried that and it didn't work. The outer build complains when that the tools project doesn't reference `net461`. This works ™️ so I'll leave it be